sdk/core/System.ClientModel/gen/ModelReaderWriterContextGenerator.cs

```diff
@@ -220,6 +226,19 @@ private void ReportDiagnosticAndEmitSource(
     /// </summary>
     internal Action<ModelReaderWriterContextGenerationSpec>? OnSourceEmitting { get; init; }
 
+    public static INamedTypeSymbol? GetProxyType(ITypeSymbol typeSymbol)
+    {
+        var persistableModelProxyAttribute = typeSymbol.GetAttributes()
+            .FirstOrDefault(attr => attr.AttributeClass?.Name == "PersistableModelProxyAttribute");
```

Comment:
Should we use the type reference to get the name?

```suggestion
            .FirstOrDefault(attr => attr.AttributeClass?.Name == typeof(PersistableModelProxyAttribute).Name);
```

Reply:
The source generator cannot depend on the system.clientmodel where that type exists it has to be the other way around.  therefore we have to use the string name vs nameof or typeof